src/traces/sankey/plot.js

```diff
@@ -0,0 +1,243 @@
+    };
+}
+
+function nodeHoveredStyle(sankeyNode, d, sankey) {
+
+    if(d && sankey) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyLink')
+            .filter(relatedLinks(d))
+            .call(linkHoveredStyle.bind(0, d, sankey, false));
+    }
+}
+
+function nodeNonHoveredStyle(sankeyNode, d, sankey) {
+
+    if(d && sankey) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyLink')
+            .filter(relatedLinks(d))
+            .call(linkNonHoveredStyle.bind(0, d, sankey, false));
+    }
+}
+
+function linkHoveredStyle(d, sankey, visitNodes, sankeyLink) {
+
+    var label = sankeyLink.datum().link.label;
+
+    sankeyLink.style('fill-opacity', 0.4);
+
+    if(label) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyLink')
+            .filter(function(l) {return l.link.label === label;})
+            .style('fill-opacity', 0.4);
+    }
+
+    if(visitNodes) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyNode')
+            .filter(relatedNodes(d))
+            .call(nodeHoveredStyle);
+    }
+}
+
+function linkNonHoveredStyle(d, sankey, visitNodes, sankeyLink) {
+
+    var label = sankeyLink.datum().link.label;
+
+    sankeyLink.style('fill-opacity', function(d) {return d.tinyColorAlpha;});
+
+    if(label) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyLink')
+            .filter(function(l) {return l.link.label === label;})
+            .style('fill-opacity', function(d) {return d.tinyColorAlpha;});
+    }
+
+    if(visitNodes) {
+        ownTrace(sankey, d)
+            .selectAll('.sankeyNode')
+            .filter(relatedNodes(d))
+            .call(nodeNonHoveredStyle);
+    }
+}
+
+module.exports = function plot(gd, calcData) {
+
+    var fullLayout = gd._fullLayout;
+    var svg = fullLayout._paper;
+    var size = fullLayout._size;
+
+    var linkSelect = function(element, d) {
+        gd._hoverdata = [d.link];
+        gd._hoverdata.trace = calcData.trace;
+        Fx.click(gd, { target: true });
+    };
+
+    var linkHover = function(element, d, sankey) {
+        d3.select(element).call(linkHoveredStyle.bind(0, d, sankey, true));
+        Fx.hover(gd, d.link, 'sankey');
+    };
+
+    var linkHoverFollow = function(element, d) {
+
+        var boundingBox = element.getBoundingClientRect();
+        var hoverCenterX = boundingBox.left + boundingBox.width / 2;
+        var hoverCenterY = boundingBox.top + boundingBox.height / 2;
+
+        var tooltip = Fx.loneHover({
+            x: hoverCenterX + window.scrollX,
+            y: hoverCenterY + window.scrollY,
+            name: d3.format(d.valueFormat)(d.link.value) + d.valueSuffix,
+            text: [
+                d.link.label,
+                ['Source:', d.link.source.label].join(' '),
+                ['Target:', d.link.target.label].join(' ')
+            ].filter(renderableValuePresent).join('<br>'),
+            color: Color.addOpacity(d.tinyColorHue, 1),
```

Comment:
This will have to be generalized for https://github.com/plotly/plotly.js/pull/1582 either in this PR or in #1582 whichever gets merged first.